src/commands/transaction/reconstruct_transaction/mod.rs

```diff
@@ -197,9 +211,25 @@ fn action_transformation(
             )))
         }
         Action::DeployContract(deploy_contract_action) => {
+            // Unfortunately, RPC doesn't return the code for the deployed contract. Only the hash.
+            // So we need to fetch it from archive node.
+
+            let code = crate::commands::contract::download_wasm::get_code(
+                &receiver_id,
+                network_config,
+                block_reference
+            ).map_err(|e| {
+                color_eyre::Report::msg(format!("Couldn't fetch the code. Please use the archival node to fetch the code. Error: {}", e))
+            })?;
+
+            let code_hash = near_primitives::hash::CryptoHash::hash_bytes(&code);
```

Comment:
nit: it would be nice to output `code_hash` hashing result with `            tracing::info!(  ` as well, 
because this is the only missing detail in a case of a mismatch, 
all other details can be learned with `near --teach-me` flag